### ISSUE TITLE
AndroidUSB connections

### ISSUE THREAD

**Issue by renezuidhof** [superpoweredSDK/Low-Latency-Android-iOS-Linux-Windows-tvOS-macOS-Interactive-Audio-Platform#689]
TLDR: 
1. After calling `Superpowered::AndroidUSB::onConnect` on an USB device, can I use it with a 'normal' `SuperpoweredAndroidAudioIO` thread?
2. Is there a way to check if an USB device has in- AND output audio without calling `Superpowered::AndroidUSB::onConnect`?

Longer version:
For my Android app I would like to use the USB stack but only if the connected USB device supports in AND output audio, like an external audio card.

So far I did not find a way to check if an attached USB device has both input and output. Only after calling `Superpowered::AndroidUSB::onConnect` I can get this information.

After connecting and finding out the USB device only has (for example) input audio like an external USB mic I don't want to start the audio on the USB stack but just continue on the 'normal' thread. Therefor I recreate the `SuperpoweredAndroidAudioIO` thread but it doesn't pick up the attached USB mic but instead uses the internal Android device mic. I tried to call `Superpowered::AndroidUSB::onDisconnect` on the USB device but that doesn't help.
Only after manually reconnecting the USB device I can use it again (and just 'cancel' the USB permissions to make sure I'm not calling `Superpowered::AndroidUSB::onConnect`).

Is this just how Android works? I saw you mentioned something like that here: https://github.com/superpoweredSDK/Low-Latency-Android-iOS-Linux-Windows-tvOS-macOS-Interactive-Audio-Platform/issues/574

If that's the case: Is there a way to determine if an USB device has in- and output audio without calling `Superpowered::AndroidUSB::onConnect`. I couldn't find a way with the Android `UsbManager` so I am wondering if there's a 'Superpowered' way of doing this?

Thanks!

**Comment by gaborszanto** [comment]
No, unfortunately these are not possible. Android can either pass a device to your app and then your app is responsible to handle it, or it stays with Android. If the device is not owned by the app, then Superpowered can not access it. If your app owns it, then Android can not access it.

**Comment by renezuidhof** [comment]
Alright, thanks!
